api/selinuxprofile/v1alpha2/selinuxprofile_types.go

```diff
@@ -98,6 +101,30 @@ type SelinuxProfile struct {
 	Status SelinuxProfileStatus `json:"status,omitempty"`
 }
 
+func (sp *SelinuxProfile) Merge(other profilebasev1alpha1.MergeableProfile) error {
+	// TODO(jhrozek): should we be defensive about checking if other attributes match as well? (e.g. inherit)
```

Comment:
should we? The profiles to merge /should/ come from the same recording, but I guess additional hardening wouldn't hurt?

Reply:
I'd say we make this private so that other projects do not use it via the API.

Reply:
By making this private do you mean "not exposed by the API modules directly"?
I'm thinking how to make this private while still letting the merge controller use an interface and not care about what profile it is that they are merging..would something like the following work for you?

```
type MergeableProfile interface {
 merge()
}

type MergeableSeccompProfile struct {
   SeccompProfile
}

func (msp *MergeableSeccompProfile) merge(....) error {
   ...
}

type MergeableSelinuxProfile struct {
   SelinuxProfile
}

func (msp *MergeableSelinuxProfile)  merge(....) error {
   ...
}
```
this might a private API in the merge controller or somewhere else in utils/ but not exposed outside SPO...

Reply:
correction: The `Merge` method could be exported outside its module, just not in the API. I guess that would still satisfy what you asked for?

Reply:
Done, please check